index.js

```diff
@@ -244,106 +242,85 @@ const getSettings = function(options) {
-    }
+    // keep output a non-array
+    b.output = (!Array.isArray(b.output) ? changeOutput(b.output) : b.output.map(changeOutput));
 
     return b;
   };
 
   /* all rollup builds by name. note only object values will be used */
   const builds = {
+    test: makeBuild('test', {
+      output: [{
+        name: `${settings.exportName}Tests`,
+        file: 'test/dist/bundle.js',
+        format: 'iife'
+      }]
+    }),
     browser: makeBuild('browser', {
       output: [{
         name: settings.exportName,
         file: `dist/${settings.distName}.js`,
         format: 'umd'
+      }, {
+        name: settings.exportName,
+        file: `dist/${settings.distName}.min.js`,
+        format: 'umd'
       }]
     }),
-    cjs: makeBuild('module', {
+    module: makeBuild('module', {
       output: [{
         file: `dist/${settings.distName}.cjs.js`,
         format: 'cjs'
-      }]
-    }),
-    es: makeBuild('module', {
-      output: [{
+      }, {
         file: `dist/${settings.distName}.es.js`,
         format: 'es'
       }]
-    }),
-    test: makeBuild('test', {
-      input: settings.testInput,
-      output: [{
-        name: `${settings.exportName}Tests`,
-        file: 'test/dist/bundle.js',
-        format: 'iife'
-      }]
-    }),
-    minBrowser: makeBuild('browser', {
-      output: [{
-        name: settings.exportName,
-        file: `dist/${settings.distName}.min.js`,
-        format: 'umd'
-      }],
-      plugins: settings.plugins.browser
-        .filter((p) => !(/^babel|uglify$/).test(p) && p !== settings.primedPlugins.babel && p !== settings.primedPlugins.uglify)
-        .concat([settings.primedPlugins.uglify, settings.primedPlugins.babel])
     })
   };
 
-  /* to prevent going into a screen during rollup */
-  process.stderr.isTTY = false;
-
-  /**
-   * if checkWatch is true, we check if rollup
-   * is used during watch. If it is we remove the minBrowser build
-   */
-  if (settings.checkWatch) {
-    for (let i = 0; i < process.argv.length; i++) {
-      if ((/^-w|--watch$/).test(process.argv[i])) {
-        delete builds.minBrowser;
-        // remove istanbul from test building during watch
-        if (settings.primedPlugins.istanbul && builds.test.plugins.indexOf('istanbul') !== -1) {
-          builds.test.plugins.splice(builds.test.plugins.indexOf(settings.primedPlugins.istanbul), 1);
-        }
-
-        break;
-      }
+  // if checkWatch is true and -w or --watch was passed to rollup
+  if (settings.checkWatch && process.argv.some((arg) => (/^-w|--watch$/).test(arg))) {
+    // remove .min.js output
+    delete builds.browser.output[1];
+
+    const testBuild = builds.test;
+    const istanbulIndex = testBuild.plugins.indexOf(settings.primedPlugins.istanbul);
+
+    // remove istanbul from test building during watch
+    if (settings.primedPlugins.istanbul && istanbulIndex !== -1) {
+      testBuild.plugins.splice(istanbulIndex, 1);
     }
   }
 
+  if (process.env.TEST_BUNDLE_ONLY) {
```

Comment:
Added environment variables that can control build outputs. 

* `TEST_BUNDLE_ONLY` should be set in `pretest` so that we only create the `test/dist/bundle.js` when testing. It also takes priority over `NO_TEST_BUNDLE` so that `pretest` can still use the `build` script.
* `NO_TEST_BUNDLE` create everything except for the test bundle. This should be set via rollup using `--environment NO_TEST_BUNDLE`. It will not be used if `TEST_BUNDLE_ONLY` is specified.

Reply:
This is to be able to skip test builds when doing stuff like publishing?

Reply:
Being able to skip `test` builds when publishing and being able to skip normal js builds when doing `npm run test` yeah